ui/StatusQ/include/StatusQ/singlemodelitemdata.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <QAbstractItemModel>
+#include <QPointer>
+#include <QQmlEngine>
+#include <QQmlPropertyMap>
+
+class SingleModelItemData : public QObject
+{
+    Q_OBJECT
+    ////////////// input
+    // the source model to get the item from
+    Q_PROPERTY(QAbstractItemModel* sourceModel READ sourceModel WRITE setSourceModel NOTIFY sourceModelChanged REQUIRED)
+    // the key role used to search for the item
+    Q_PROPERTY(QString key READ key WRITE setKey NOTIFY keyChanged REQUIRED)
+    // the value role used to cache the item
+    Q_PROPERTY(QVariant value READ value WRITE setValue NOTIFY valueChanged REQUIRED)
+    // whether to cache the item when it is removed from the model
+    // if true, the item will be cached and available until another source model is used or the cacheOnRemoval is set to false
+    Q_PROPERTY(bool cacheOnRemoval READ cacheOnRemoval WRITE setCacheOnRemoval NOTIFY cacheOnRemovalChanged)
+
+    ///////////// output
+    // the item found in the source model
+    Q_PROPERTY(QQmlPropertyMap* modelItem READ item NOTIFY itemChanged)
+    // whether the item is available
+    Q_PROPERTY(bool available READ available NOTIFY availableChanged)
+    // the roles of the item
+    Q_PROPERTY(QStringList roles READ roles NOTIFY rolesChanged)
+    // the row of the item in the source model
+    Q_PROPERTY(int row READ row NOTIFY rowChanged)
```

Comment:
I'd favor `row` since the meaning is explicit; with `index` I never know if it's the row number or a `QModelIndex`

Reply:
> I'd favor `row` since the meaning is explicit; with `index` I never know if it's the row number or a `QModelIndex`

yes, this is how I ended up naming it `row`. Initially I wanted to expose the`QModelIndex`, but changed my mind and changed it to row 😄

Reply:
So, we have mixed preferences here 😄 
I'll leave it as it is if there's no objection.